File: 724-find-pivot-index/find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
        int sum = 0;

        for(int ele : nums) {
            sum += ele;
        }
        int r_sum = sum;
         int l_sum = 0;

         for(int i = 0; i <= nums.length-1; i++) {
            r_sum -= nums[i];

            if(r_sum == l_sum) {
                return i;
            }
            else {
                l_sum += nums[i];
            }
        }
        return -1;
    }
}

File: 724-find-pivot-index/find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        n = len(nums)
        ps = [0] * n
        ps[0] = nums[0]

        for i in range(1, n):
            ps[i] = ps[i - 1] + nums[i]

        totalSum = ps[n - 1]

        for i in range(n):

            if i == 0:
                leftSum = 0
            else:
                leftSum = ps[i - 1]

            rightSum = totalSum - ps[i]

            if leftSum == rightSum:
                return i

        return -1